src/datasets/keyhash.py

```diff
@@ -70,11 +70,17 @@ def __init__(self, hash_data):
 class DuplicatedKeysError(Exception):
     """Raise an error when duplicate key found."""
 
-    def __init__(self, key):
-        self.prefix = "FAILURE TO GENERATE DATASET !"
-        self.err_msg = f"\nFound duplicate Key: {key}"
-        self.suffix = "\nKeys should be unique and deterministic in nature"
-        super().__init__(f"{self.prefix}{self.err_msg}{self.suffix}")
+    def __init__(self, key, duplicate_key_indices=None):
+        if duplicate_key_indices:
+            self.prefix = f"Found multiple examples with duplicate key: {key}"
+            self.err_msg = f"\nThe following examples {' ,'.join(duplicate_key_indices)} have the same key {key} "
+            self.suffix = "\nPlease fix the dataset script at <Path to Dataset>"
+            super().__init__(f"{self.prefix}{self.err_msg}{self.suffix}")
+        else:
+            self.prefix = "FAILURE TO GENERATE DATASET !"
+            self.err_msg = f"\nFound duplicate Key: {key}"
+            self.suffix = "\nKeys should be unique and deterministic in nature"
+            super().__init__(f"{self.prefix}{self.err_msg}{self.suffix}")
```

Comment:
I think you can always require `duplicate_key_indices` and completely drop the old `FAILURE TO GENERATE DATASET` message

Reply:
One nit by me: 

I like this error message format better:

```
DuplicateKeysError: both 42nd and 1337th examples have the same key `48`.
Please fix the dataset script at <path/to/the/dataset/script>
```

(Originally proposed by @lhoestq in the referenced issue)

PS: The absolute indices needed for this format can be computed using ArrowWriter's `_num_examples` attribute.

Reply:
@mariosasko  , Apologies , I saw your comment after I pushed https://github.com/huggingface/datasets/pull/4545/commits/ccba4f500288f78824bc638b944af772f99c55de. 

The current Error looks like : 

```
DuplicatedKeysError: Found multiple examples generated with the same key
The following examples 15, 8784 have the key 519
Please fix the dataset script at datasets/wikicorpus/wikicorpus.py to avoid duplicate keys
```

Do let me know if you still prefer the original proposal and I shall update it.

```
DuplicateKeysError: both 42nd and 1337th examples have the same key `48`.
Please fix the dataset script at <path/to/the/dataset/script>
```

Also could you elaborate on what you mean by absolute indices can be computed using `_num_examples` attribute ? 
Should the indices by `_num_examples + index` ?